### PR TITLE
ORC-976: Optimize compute zigzagliterals

### PR DESCRIPTION
**Pull request by guiyanakuang**
### What changes were proposed in this pull request?

```java
  private void computeZigZagLiterals() {
    // populate zigzag encoded literals
    long zzEncVal = 0;
    for (int i = 0; i < numLiterals; i++) {
      if (signed) {
        zzEncVal = utils.zigzagEncode(literals[i]);
      } else {
        zzEncVal = literals[i];
      }
      zigzagLiterals[i] = zzEncVal;
    }
  }
```

Avoid conditional judgments in loops. 
The unsigned case can use literals instead of zigzagLiterals. 

### Why are the changes needed?

optimize write performance.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you, @guiyanakuang . In this case, we had batter improve C++ code together for consistency. Could you update it in this PR?

https://github.com/apache/orc/blob/main/c++/src/RleEncoderV2.cc#L170-L180

cc @wgtmac

File: java/core/src/java/org/apache/orc/impl/RunLengthIntegerWriterV2.java
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/RunLengthIntegerWriterV2.java:550]: Shall we remove `if` statement and add an assertion at the first line of this function?

**Comment by dongjoon-hyun** [comment]
cc @pgaref and @williamhyun , too.

File: java/core/src/java/org/apache/orc/impl/RunLengthIntegerWriterV2.java
Comment (guiyanakuang) [reply java/core/src/java/org/apache/orc/impl/RunLengthIntegerWriterV2.java]: I quite agree. fix in 7de084b

**Comment by guiyanakuang** [comment]
> Thank you, @guiyanakuang . In this case, we had batter improve C++ code together for consistency. Could you update it in this PR?
> 
> https://github.com/apache/orc/blob/main/c++/src/RleEncoderV2.cc#L170-L180
> 
> cc @wgtmac

I think I can, although I don't use c++ much

**Comment by dongjoon-hyun** [comment]
Thank you for updating and considering C++ change together, @guiyanakuang .

**Comment by guiyanakuang** [comment]
Update C++ improve, please help review code. I don't use C++ much. @dongjoon-hyun @wgtmac

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM from my side. Thank you so much for updating, @guiyanakuang .
I'll wait for the other committers' review.

**Comment by dongjoon-hyun** [comment]
Also, cc @omalley , too

**Review by wgtmac — APPROVED** [review]
LGTM +1. Thanks @guiyanakuang and @dongjoon-hyun !

**Comment by dongjoon-hyun** [comment]
Thank you, @guiyanakuang and @wgtmac !
Merged to main/1.7.

**Pull request merged**
